pkg/stores/proxy/proxy_store.go

```diff
@@ -422,20 +422,29 @@ func (s *Store) Create(apiOp *types.APIRequest, schema *types.APISchema, params
 	}
 
 	name := types.Name(input)
-	ns := types.Namespace(input)
-	if name == "" && input.String("metadata", "generateName") == "" {
-		input.SetNested(schema.ID[0:1]+"-", "metadata", "generatedName")
+	namespace := types.Namespace(input)
+	generateName := input.String("metadata", "generateName")
+
+	if name == "" && generateName == "" {
+		input.SetNested(schema.ID[0:1]+"-", "metadata", "generateName")
 	}
-	if ns == "" && apiOp.Namespace != "" {
-		ns = apiOp.Namespace
-		input.SetNested(ns, "metadata", "namespace")
+
+	if namespace == "" {
+		if apiOp.Namespace == "" {
```

Comment:
@tomleb pointed out that this needs to consider cluster-scoped resources as well (so things like cluster roles). In those cases, you won't have a namespace or an apiOp.Namespace, but that is fine since those resources don't go in a namespace. I would take a look at `attributes.Namespaced(schema)` to see if that can give you sufficient information here.

Reply:
@MbolotSuse Added attributes.Namespaced check before this. Also added tests for Create function.